src/lib.rs

```diff
@@ -1072,6 +1072,98 @@ impl Graphics2D
         );
     }
 
+    /// Draws a single-color rounded rectangle at the specified location. The
+    /// coordinates of the rounded rectangle are specified in pixels.
+    #[inline]
+    pub fn draw_rounded_rectangle(
+        &mut self,
+        round_rect: impl AsRef<RoundedRectangle>,
+        color: Color
+    )
+    {
+        let round_rect = round_rect.as_ref();
+
+        //create polygon starting clockwise from top-left cornet
+        let polygon = Polygon::new(&[
```

Comment:
Polygons are very expensive to draw (mostly during the initial creation stage) -- it would be better to draw three rectangles (left, middle, and right, where the middle one is taller).

Reply:
On it; btw: aren't polygons just a bunch of triangles?

Reply:
That's correct, however the `Polygon` constructor has to calculate where those triangles are, using a process known as earcutting. This also involves allocating memory (`Vec`).